### ISSUE TITLE
Question about Debiased Inbatch Loss

### ISSUE THREAD

**Issue by rabbicat30** [xyliugo/ODMT#4]
1. The explicit frequency calculation in the code is equal to the number of occurrences divided by the sum of the number of items in the training set, which is a number less than 1. After taking the log, it is a negative number, and after subtracting this negative number, the logits become bigger. Is this a misunderstanding on my part? If yes can you explain how this place works?
2. If the logits are calculated on all items, but the training set contains only some of the items, are the debias_logits of the items that do not appear in the training set set set to 0?

**Comment by xyliugo** [comment]
1. The debiased in-batch loss assigns lower weight to more frequent items and higher weight to less frequent ones. For detailed explanations, see: 1) [XIR](https://arxiv.org/pdf/2205.14859.pdf); 2) [Google Debias](https://research.google/pubs/sampling-bias-corrected-neural-modeling-for-large-corpus-item-recommendations/).

2. It seems correct, but I'm unsure if the count is from the training set or the entire dataset. You can check the code for clarification.
